### PR TITLE
Get: userlist and search

### PR DESCRIPTION
**Pull request by tyxxxx**
Hi All, 
Now, the filtering function is already working. When you open the localhost, all users and posts in db will load up. When u click users in the userlist, the corresponding posts will be filtered. You can also filter price, location and category multiple times by entering the value. Clicking the cancel key will take you back to the homepage

### CODE REVIEW

**Comment by BarneyShi** [comment]
@tyxxxx I think you first need to change the base branch to `project_4`. It's hard to tell what code were already there and what code are new.

**Comment by NaithanB** [comment]
Hi @tyxxxx, I'm just starting to look through this now in more detail. I like the user filtering. The user icons look nicer now. It would be nice if there was some visual feedback to show which user is currently selected. Maybe something like highlighting the name.

I think I found a few bugs that should probably be fixed before we merge. The first bug is in the price filter. The first time I enter 0 to 1000 and hit submit, I get back all the posts in the database (even the ones outside of that price range). When I try it a second time, I get back only 1 result but there should be more than 1 post in that price range in the database. I notice too that when I filter price by entering "from 2001 to 999" and hit submit, I get back four posts with prices 900, 900, 800, and 7, which is outside of the search range. Are the search results working that way on your end too?

There seems to be a get request that gets all of the posts whenever I select a filter in the searchbar. I think the only time that we need to do a get request for all the posts should be when we reload the page or possibly also when we hit the cancel button. We can use the get request I wrote to handle that.

The cancel button seems to be broken on my end. When I switch to one of the filters and hit cancel, it doesn't return to the user display.

I think it would be a good idea to use projection to make sure we only return the info that needs to be displayed on the main page. If you look at the /posts get request in index.js on my branch, there's a projection you could use that only returns one of the images instead of the full array of images, which could reduce the transfer time a lot.

You mentioned that there will be some merge issues with between our two branches. I think you might be right, but even though we're working in a couple of the same files, I think most of our code is in separate blocks, so hopefully it won't be too bad.

**Comment by NaithanB** [comment]
@BarneyShi, if you're using the github site to view the diff, if you choose the latest commit on this branch I think you can see the changes that way for the time being. But yeah, we don't want to merge into the main branch.

**Comment by tyxxxx** [comment]
@NaithanB Hi Naithan, It seems that there is a bug in the price filtering. I will fix it, but there is no problem with the cancel key. Would you like to test it again? Maybe because the data is large and the loading time is long, I need to wait a little while to return to the main interface when I click the cancel key here

**Comment by tyxxxx** [comment]
@NaithanB Hi Naithan, It seems that there is a bug in the price filtering. I will fix it, but there is no problem with the cancel key. Would you like to test it again? Maybe because the data is large and the loading time is long, I need to wait a little while to return to the main interface when I click the cancel key here

**Comment by NaithanB** [comment]
Ah, yeah, you're right about the cancel button. That's definitely what the problem was. Hmm... it took about 20 seconds for cancel to complete. That's quite long. Once we integrate a loading spinner it will be clearer that something is happening on click, but if we never click submit I wonder if we could avoid doing the fetch all together.

File: vanhouse/src/components/Price.js [hunk 1]
```diff
@@ -1,42 +1,97 @@
...
+
+
+  function Cancel() {
+
+    fetch("http://localhost:4000/getpost", { method: "GET" })
+      .then((res) => res.json())
+      .then((res) => {
+        console.log(res);
+        setFilterPost(res);
+    setState(1);
```
Comment (BarneyShi) [inline vanhouse/src/components/Price.js:20]: @tyxxxx I think it's a better idea to just store all posts in a variable, and then you just need to call `setFilterPost(allPosts)` when users click cancel button. In this way we don't need to call backend and it's much faster. Same to other cancel operations.

File: vanhouse/src/components/Price.js [hunk 2]
```diff
@@ -1,42 +1,97 @@
...
+      });
+  }
+
+  function getP() {
+    // const rr = [];
+    // storePost.forEach((item) => {
+    //   if (item.price >= low && item.price <= high) {
+    //     rr.push(item);
+    //   }
+    // });
```
Comment (BarneyShi) [inline vanhouse/src/components/Price.js:31]: Remove commented code.

File: vanhouse/src/components/UserList.js [hunk 3]
```diff
@@ -1,103 +1,90 @@
...
-import user2 from '../assets/img2.jpg';
-import user3 from '../assets/img3.jpg';
+import PropTypes from "prop-types";
+import React, { useEffect, useState } from "react";
+import { Button, Col, Container, Row } from "react-bootstrap";
+import user1 from "../assets/img1.jpg";
+import user2 from "../assets/img2.jpg";
+import user3 from "../assets/img3.jpg";
 
+function UserList({ setFilterPost }) {
```
Comment (BarneyShi) [inline vanhouse/src/components/UserList.js:8]: @tyxxxx I found we can only see all posts by refreshing page after clicking on a username. We need to find a way to see all posts again.

**Review by BarneyShi — CHANGES_REQUESTED** [review]

File: backend/routes/index.js [hunk 4]
```diff
@@ -1,9 +1,142 @@
...
+    res.json(data);
+    data.forEach(item => {
+      User.findOne({id: item.authorID}, (err, da) => {
+        console.log(da);
+      })
+    })
+  });
+});
+
+router.get("/price", function (req, res, next) {
```
[the remark this answers is not in the document]
Comment (tyxxxx) [reply backend/routes/index.js:78]: For this one, I think it's better to keep three filters now. Just like the screening of air tickets, some people don't care about the price, they only care about the location; Some people just care about the address, not the price. I also refer to Kijiji for this filter. There are separate filters and merged filters

[the remark this answers is not in the document]
Comment (BarneyShi) [reply backend/routes/index.js:78]: @tyxxxx That sounds good. 👍

**Comment by tyxxxx** [comment]
I made some changes. I added the cancel button in user list and it will return to the homepage. When you click other cancel buttons,  they will return to the homepage immediately, Then take a moment to load all the posts. During merge, Naithan will handles the conflict in postCollection? @NaithanB 
pls check this one @BarneyShi @parksj23

**Pull request closed without merging**